### PR TITLE
fix: Various kernel fixes

### PR DESCRIPTION
**Pull request by trungnt2910**
Various fixes are made in order to ~~run `dotnet` properly~~ make `dotnet` half-runnable.

## proc_info:
- Added 3 additional flavors.
- Added more info for `_proc_pidinfo_tbsdinfo`

## waitid
- Marshalled the `siginfo_t` between Linux and BSD, allowing processes to access valid data.

## SIGCHLD
- Some processes check for explicitly ignored signals and avoid handling them (e.g. `dotnet`). As `SIGCHLD` is not configured with `SIG_IGN` on real MacOS, we should not do that on Darling, either.

### CODE REVIEW

**Comment by LubosD** [comment]
LGTM. Anyone else wants to review?

**Review by facekapow — APPROVED** [review]
LGTM. My only concern was the `SIG_IGN` change affecting LLDB, but doing some testing with an old LLDB version (one that was actually working) shows that it still works fine.

**Comment by trungnt2910** [comment]
@facekapow @LubosD Can you review the last commit? It's simply some reexported symbols in `CoreServices` to get some `dlsym`s working.

File: src/frameworks/CoreServices/reexport.exp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+_log2$fenv_access_off
+_modff
+_modff$fenv_access_off
+_nan
+_nanf
+_nextafterd
+_nextafterf
+_remquo
+_remquo$fenv_access_off
+dyld_stub_binder
```
Comment (facekapow) [inline src/frameworks/CoreServices/reexport.exp:33]: `dyld_stub_binder` isn't supposed to be re-exported; it's automatically added by `ld` for each library.
```suggestion
```

**Review by facekapow — CHANGES_REQUESTED** [review]
Just a small issue with re-exporting `dyld_stub_binder`. Other than that, LGTM.

File: src/frameworks/CoreServices/reexport.exp (hunk 1 above)
Comment (trungnt2910) [reply src/frameworks/CoreServices/reexport.exp:33]: Done. Forced pushed the change.

Comment (trungnt2910) [reply src/frameworks/CoreServices/reexport.exp]: It's a bit of an oversight when copy-pasting stuff.

**Review by facekapow — APPROVED** [review]

**Comment by trungnt2910** [comment]
Thanks so much! This pull request wouldn't have been possible without your mentoring!

**Pull request merged**